src/devices/src/virtio/block/request.rs

```diff
@@ -178,67 +178,72 @@ impl Request {
         Ok(req)
     }
 
+    fn execute_seek(&self, disk: &mut DiskProperties) -> result::Result<(), ExecuteError> {
+        // Check that the data length is a multiple of 512 as specified in the virtio standard.
+        // We could also check this at request parsing level.
+        if u64::from(self.data_len) & (SECTOR_SIZE - 1) != 0 {
```

Comment:
same, I think this check is erroneous. It doesn't return true for 513 for example

Reply:
I feel like these checks don't belong in the `execute_seek` method. I think they are more suited in the `Request` parsing logic, WDYT?